### PR TITLE
tests: test_xfail_handling: use sys.dont_write_bytecode

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by bluetech** [comment]
One thing to worry about when modifying globals like `sys.dont_write_bytecode` even temporarily is that it also affects other concurrent threads. I guess no one runs pytest in multiple threads though, right? It doesn't make much sense with current python. If so, this looks like a nice cleanup to me (though @nicoddemus added this test so they might want to approve).

**Comment by blueyed** [comment]
@bluetech 
> sys.dont_write_bytecode

We're using it elsewhere already.  See 1abb08d52.

File: testing/test_stepwise.py [hunk 1]
```diff
@@ -167,11 +167,13 @@ def test_stop_on_collection_errors(broken_testdir, broken_first):
     result.stdout.fnmatch_lines("*error during collection*")
 
 
-def test_xfail_handling(testdir):
+def test_xfail_handling(testdir, monkeypatch):
     """Ensure normal xfail is ignored, and strict xfail interrupts the session in sw mode
 
     (#5547)
     """
+    monkeypatch.setattr("sys.dont_write_bytecode", True)
```
Comment (RonnyPfannschmidt) [inline testing/test_stepwise.py]: does this still work if one forces subprocess execution? if not you may need to add the env var monkeypatch as well

Comment (blueyed) [reply testing/test_stepwise.py]: Likely not (as with other places we're doing it).
Might be worth having a method in monkeypatch for handling this in a canonical way, but I think we do not support using `--runpytest=subprocess` in our test suite currently.

Comment (RonnyPfannschmidt) [reply testing/test_stepwise.py]: good point, lets make a follow-up issue to later decide the  future of that

**Comment by blueyed** [comment]
@nicoddemus 
I would assume this is good to go, please ACK.

**Pull request merged**